### PR TITLE
'package: ^version' copyable title pattern on package page.

### PR DESCRIPTION
**Pull request by isoos**
- The most requested part of #3336.

- with selection:
  Screenshot 2020-10-29 at 10 11 31

- only hover:
  Screenshot 2020-10-29 at 10 10 59

- without hover:
  Screenshot 2020-10-29 at 10 11 06

### CODE REVIEW

**Comment by isoos** [comment]
The light-gray `: ^` part is displayed only on hovering, with a short transition.
/cc @mit-mit

**Comment by isoos** [comment]
Alternatively we could go with a copy "clipboard" dropdown, after the title?

**Comment by sigurdm** [comment]
> Alternatively we could go with a "copy to clipboard" icon + dropdown, after the title?

I think that would have much better discoverability - perhaps we could have both, so when you hover the icon, it highlights what you are going to copy.

File: app/lib/frontend/templates/detail_page.dart [hunk 1]
```diff
@@ -26,9 +27,16 @@ String renderDetailHeader({
   /// Set true for more whitespace in the header.
   bool isLoose = false,
 }) {
+  if (title == null && titleHtml == null) {
+    throw ArgumentError('One of `title` and `titleHtml` must be specified.');
+  }
+  if (title != null && titleHtml != null) {
+    throw ArgumentError(
+        'Only one of `title` and `titleHtml` must be specified.');
+  }
```
Comment (sigurdm) [inline app/lib/frontend/templates/detail_page.dart:36]: nit
```suggestion
  if ((title == null && titleHtml == null) || (title != null && titleHtml != null)) {
    throw ArgumentError(
        'Exactly one of `title` and `titleHtml` must be specified.');
  }
```

**Comment by isoos** [comment]
Closing in favour of #4201

**Pull request closed without merging**